modules/dmrpp_module/Chunk.cc

```diff
@@ -612,23 +617,127 @@ void Chunk::filter_chunk(const string &filters, unsigned long long chunk_size, u
 
     vector<string> filter_array = BESUtil::split(filters, ' ' );
 
+    // We need to check if the filters that include the deflate filters are contiguous.
+    // That is: the filters must be something like "deflate deflate deflate" instead of "deflate other_filter deflate"
+
+    bool is_1st_deflate = true;
+    unsigned cur_deflate_index = 0;
+    unsigned num_deflate = 0;
+
+    for (unsigned i = 0; i<filter_array.size(); i++) {
+
+        if (filter_array[i] == "deflate") {
+            if (is_1st_deflate == true) {
+                cur_deflate_index = i;
+                is_1st_deflate = false;
+            }
+            else if (i != (cur_deflate_index+1)) {
+                throw BESInternalError("The deflate filters must be adjacent to each other",
+                                       __FILE__, __LINE__);
+            }
+            else 
+                cur_deflate_index = i;
+            num_deflate++;
+        }
+
+    }
+
+    // If there are >1 deflate filters applied, we want to localize the handling of the compressed data  in this function.
+    unsigned deflate_index = 0;
+    unsigned long long out_buf_size = 0;
+    unsigned long long in_buf_size = 0;
+    char**destp = nullptr;
+    char* dest = nullptr;
+#if 0
+    char* tmp_buf = nullptr;
+#endif
+    char* tmp_dest = nullptr;
+
+    bool ignore_rest_deflate = false;
+
     for (auto i = filter_array.rbegin(), e = filter_array.rend(); i != e; ++i) {
+
         string filter = *i;
 
         if (filter == "deflate") {
-            char *dest = new char[chunk_size];
-            try {
-                inflate(dest, chunk_size, get_rbuf(), get_rbuf_size());
-                // This replaces (and deletes) the original read_buffer with dest.
+
+            // Here we find that the deflate filter is applied twice. 
```

Comment:
Is this because the premise of the DMR++ document that all chunks for a given variable/dataset have the same filters is wrong? If so, we should probably fix that. Let me know if that's the case and we can make tickets for it, et cetera.